apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/codegen/kotlin/EnumType.kt

```diff
@@ -4,19 +4,32 @@ import com.apollographql.apollo.compiler.applyIf
 import com.apollographql.apollo.compiler.ast.EnumType
 import com.squareup.kotlinpoet.*
 
-internal fun EnumType.typeSpec(generateAsInternal: Boolean = false) =
-    TypeSpec
-        .enumBuilder(name)
-        .applyIf(description.isNotBlank()) { addKdoc("%L\n", description) }
-        .applyIf(generateAsInternal) { addModifiers(KModifier.INTERNAL) }
-        .primaryConstructor(primaryConstructorSpec)
-        .addProperty(rawValuePropertySpec)
-        .apply {
-          values.forEach { value -> addEnumConstant(value.constName, value.enumConstTypeSpec) }
-          addEnumConstant("UNKNOWN__", unknownEnumConstTypeSpec)
-        }
-        .addType(companionObjectSpec)
-        .build()
+internal fun EnumType.typeSpec(
+    generateAsInternal: Boolean = false,
+    enumAsSealedClassPatternFilters: List<Regex>
+): TypeSpec {
+  val asSealedClass = enumAsSealedClassPatternFilters.isNotEmpty() && enumAsSealedClassPatternFilters.any { pattern ->
```

Comment:
Is the check for `isNotEmpty()` needed here ? I would expect `any` to return false on empty lists.